### ISSUE TITLE
Style/NestedModifier auto-corrects into invalid ruby code

### ISSUE THREAD

**Issue by dmitryv** [rubocop-hq/rubocop#7358]
`puts a unless [1,2].include? a if a`

gets converted into

`puts a if a && ![1,2].include? a`

, which has a syntax error.

--------

## Steps to reproduce the problem

test.rb:

```
a = 1
puts a unless [1,2].include? a if a
```

```
rubocop --only Style/NestedModifier --auto-correct ./test.rb
Inspecting 1 file
E

Offenses:

test.rb:2:8: C: [Corrected] Style/NestedModifier: Avoid using nested modifiers.
puts a unless [1,2].include? a if a
       ^^^^^^
test.rb:2:32: E: Lint/Syntax: unexpected token tIDENTIFIER
(Using Ruby 2.3 parser; configure using TargetRubyVersion parameter, under AllCops)
puts a if a && ![1,2].include? a
                               ^

1 file inspected, 2 offenses detected, 1 offense corrected
```

test.rb after correction:

```
a = 1
puts a if a && ![1,2].include? a
```

```
ruby test.rb
test.rb:2: syntax error, unexpected tIDENTIFIER, expecting end-of-input
```

## RuboCop version

```
0.74.0 (using Parser 2.6.4.1, running on ruby 2.3.3 x86_64-darwin18)

.rubocop.yml
AllCops:
  TargetRubyVersion: 2.3
```

**Comment by Drenmi** [comment]
Looked into this briefly. It seems that method sends that appear as the non-first expression in a boolean logic expression must be parenthesized:

**Okay:**

```ruby
foo && foo(bar)
```

**Syntax error:**

```ruby
foo && foo(bar)
```
